Moving invenio to bootstrap 3 - Makefile updated

kneczaj opened the pull request.

kneczaj committed:
Style: migration to twitter bootstrap 3

 * fast replacement - accordion, accordion-toggle deleted
 * header porting completed
 * searchbar migration
 * fixed different padding at different window widths
 * css fom page.html, header.html moved to base.css
 * layout of the search page adapted
 * footer stylization
 * portalboxes migrated to panels
 * registration, login page form migration
 * icon -> glyphicon replacement
 * account management page migrated
 * navarheader substituted with dropdown-header in a couple of places
 * search settings menu improved
 * advanced search form adapted
 * replaced all occurences of buttons-radio -> buttons
 * searchbar and search options menu improvements
 * search settings layout improved
 * makefile modified to install bootstrap3

jirikuncar commented:
Can you rebase and squash your commits?

kneczaj commented:
finished

jalavik commented:
Also this file should be removed: invenio/modules/workflows/static/js/workflows/DT_bootstrap.js

jirikuncar commented:
Please, keep this branch up to date. I will try to merge it in few days. There is quite a lot of changes to be check. Thanks!

jirikuncar commented:
Can you please check that `/record/<id>/comments` are working properly with the Bootstrap 3 so @adrianp can work after this branch is merged?

jirikuncar commented:
- [ ] add to search dropdown is broken
- [ ] messages menu is too small
- [ ] deposit forms are not properly displayed
- [ ] messages filter form is not wrapped
- [ ] make install-bootstrap - create font directory first!

kneczaj commented:
> > > > Can you please check that /record/<id>/comments are working properly
> > > > with the Bootstrap 3 so@adrianp https://github.com/adrianp can work after
> > > > this branch is merged?

Can you tell me how to reach this url? or what sample id to type there?

> > > > add to search dropdown is broken

Do you mean the carret button on right of the search input field? I have no
idea where to place the script to make it work.

> > > > messages menu is too small & messages filter form is not wrapped
> > > > Agree - I totally ommited this
> > > > 
> > > > deposit forms are not properly displayed

I have rebase my bootstrap3 branch to the current
code-structure-with-webdeposit. Unfortunately, deposit page doesn't work
for me at all. I've pushed the merge back to my branch.

I have this error:

Error in registering exception to
'/home/kamil/cern/virtualenvs/bootstrap3/var/log/invenio.error': 'Neither
'now' object nor 'Comparator' object has an attribute 'strftime''
127.0.0.1 - - [21/Nov/2013 14:08:49] "GET /deposit/ HTTP/1.1" 500 -
Traceback (most recent call last):
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask/app.py",
line 1836, in **call**
    return self.wsgi_app(environ, start_response)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/src/invenio/invenio/ext/legacy/**init**.py",
line 62, in **call**
    response = self.app.handle_exception(e)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask/app.py",
line 1403, in handle_exception
    reraise(exc_type, exc_value, tb)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/src/invenio/invenio/ext/legacy/**init**.py",
line 58, in **call**
    response = self.app.full_dispatch_request()
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask/app.py",
line 1477, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask/app.py",
line 1381, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask/app.py",
line 1475, in full_dispatch_request
    rv = self.dispatch_request()
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask/app.py",
line 1461, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/flask_login.py",
line 658, in decorated_view
    return func(_args, *_kwargs)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/src/invenio/invenio/modules/deposit/views/deposit.py",
line 103, in index
    my_depositions=Deposition.get_depositions(current_user),
  File
"/home/kamil/cern/virtualenvs/bootstrap3/src/invenio/invenio/modules/deposit/models.py",
line 950, in get_depositions
    BibWorkflowObject.modified.desc()).all()
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2241,
in all
    return list(self)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2353,
in **iter**
    return self._execute_and_instances(context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/orm/query.py", line 2368,
in _execute_and_instances
    result = conn.execute(querycontext.statement, self._params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 662,
in execute
    params)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 761,
in _execute_clauseelement
    compiled_sql, distilled_params
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 874,
in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 1024,
in _handle_dbapi_exception
    exc_info
  File "build/bdist.linux-x86_64/egg/sqlalchemy/util/compat.py", line 196,
in raise_from_cause
    reraise(type(exception), exception, tb=exc_tb)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/base.py", line 867,
in _execute_context
    context)
  File "build/bdist.linux-x86_64/egg/sqlalchemy/engine/default.py", line
324, in do_execute
    cursor.execute(statement, parameters)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/MySQLdb/cursors.py",
line 201, in execute
    self.errorhandler(self, exc, value)
  File
"/home/kamil/cern/virtualenvs/bootstrap3/lib/python2.7/site-packages/MySQLdb/connections.py",
line 36, in defaulterrorhandler
    raise errorclass, errorvalue
OperationalError: (OperationalError) (1054, "Unknown column
'bwlOBJECT._extra_data' in 'field list'") 'SELECT `bwlWORKFLOW`.uuid AS
`bwlWORKFLOW_uuid`, `bwlWORKFLOW`.name AS `bwlWORKFLOW_name`,
`bwlWORKFLOW`.created AS `bwlWORKFLOW_created`, `bwlWORKFLOW`.modified AS
`bwlWORKFLOW_modified`, `bwlWORKFLOW`.id_user AS `bwlWORKFLOW_id_user`,
`bwlWORKFLOW`.extra_data AS `bwlWORKFLOW_extra_data`, `bwlWORKFLOW`.status
AS `bwlWORKFLOW_status`, `bwlWORKFLOW`.current_object AS
`bwlWORKFLOW_current_object`, `bwlWORKFLOW`.counter_initial AS
`bwlWORKFLOW_counter_initial`, `bwlWORKFLOW`.counter_halted AS
`bwlWORKFLOW_counter_halted`, `bwlWORKFLOW`.counter_error AS
`bwlWORKFLOW_counter_error`, `bwlWORKFLOW`.counter_finished AS
`bwlWORKFLOW_counter_finished`, `bwlWORKFLOW`.module_name AS
`bwlWORKFLOW_module_name`, `bwlOBJECT`.id AS `bwlOBJECT_id`,
`bwlOBJECT`._data AS `bwlOBJECT__data`, `bwlOBJECT`._extra_data AS
`bwlOBJECT__extra_data`, `bwlOBJECT`.id_workflow AS
`bwlOBJECT_id_workflow`, `bwlOBJECT`.version AS `bwlOBJECT_version`,
`bwlOBJECT`.id_parent AS `bwlOBJECT_id_parent`, `bwlOBJECT`.created AS
`bwlOBJECT_created`, `bwlOBJECT`.modified AS `bwlOBJECT_modified`,
`bwlOBJECT`.status AS `bwlOBJECT_status`, `bwlOBJECT`.data_type AS
`bwlOBJECT_data_type`, `bwlOBJECT`.uri AS `bwlOBJECT_uri`,
`bwlOBJECT`.id_user AS `bwlOBJECT_id_user` \nFROM `bwlOBJECT` INNER JOIN
`bwlWORKFLOW` ON `bwlWORKFLOW`.uuid = `bwlOBJECT`.id_workflow \nWHERE
`bwlWORKFLOW`.module_name = %s AND `bwlWORKFLOW`.id_user = %s ORDER BY
`bwlOBJECT`.modified DESC' ('webdeposit', 1L)

2013/11/21 jirikuncar notifications@github.com

> -  add to search dropdown is broken
> -  messages menu is too small
> -  deposit forms are not properly displayed
> -  messages filter form is not wrapped
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jirikuncar/invenio/pull/59#issuecomment-28973440
> .

adrianp commented:
@kneczaj e.g. `http://localhost:5000/record/93/comments` (given that you have Atlantis installed); if you could make sure that this page displays correctly with Bootstrap3 it would be really helpful

kneczaj commented:
Right now the page displays correctly, but if you click "Write comment"
then you get weird black background, which stays after closing "Write
comment" form.

2013/11/21 Adrian-Tudor Panescu notifications@github.com

> @kneczaj https://github.com/kneczaj e.g.
> http://localhost:5000/record/93/comments (given that you have Atlantis
> installed); if you could make sure that this page displays correctly with
> Bootstrap3 it would be really helpful
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jirikuncar/invenio/pull/59#issuecomment-28984063
> .

adrianp commented:
@kneczaj In the current version, when you click on add comment you get this:

![comments](https://f.cloud.github.com/assets/480478/1591956/16d69304-52b2-11e3-8ef6-60ddda1067bf.png)

Clearly, when you close the dialog, the shaded background goes away. So I guess it's a Boostrap3 issue that the shaded background does not go away in your version.

kneczaj commented:
Yes, you're right, and there is no white background under the form too.
It's just draw directly on this black one - on full screen.

2013/11/21 Adrian-Tudor Panescu notifications@github.com

> @kneczaj https://github.com/kneczaj In the current version, when you
> click on add comment you get this:
> 
> [image: comments]https://f.cloud.github.com/assets/480478/1591956/16d69304-52b2-11e3-8ef6-60ddda1067bf.png
> 
> Clearly, when you close the dialog, the shaded background goes away.
> 
> So I guess it's a Boostrap3 issue that the shaded background does not go
> away.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/jirikuncar/invenio/pull/59#issuecomment-28984763
> .

kneczaj commented:
If you tell me where is the form rendering made, I can fix it.

I modified it in my browser, and the html should look like below:

<div class="modal" aria-hidden="false" style="display: block;">
   <div class="modal-dialog">
      <div class="in"></div><div class="modal-content">
<div class="modal-header">
    <a class="close" data-dismiss="modal">×</a>
    <h3>Write a comment</h3>
  </div>

<div class="modal-body">
<form name="write_message" action="/record/93/comments/add"
style="margin-bottom:0px;" method="post" class="form-horizontal">
  <div class="modal-body">
    <input id="csrf_token" name="csrf_token" type="hidden"
value="1385042454.21##2b7b496eddaf216e5d8966681d60ad87093f55c4">
  <input id="in_reply_to_id_cmtRECORDCOMMENT"
name="in_reply_to_id_cmtRECORDCOMMENT" type="hidden" value="">
  <div class="form-group ">
    <label class="control-label" for="title">Title</label>
    <input class=" form-control" id="title" name="title" type="text"
value="">
      </div>

  <div class="form-group ">

```
<label class="control-label" for="body">Message</label>



<textarea class=" form-control" id="body" name="body"></textarea>


  </div></div>
```

  </form></div>
     <div class="modal-footer  in" aria-hidden="false">
        <input type="submit" name="send_button" value="Send" class="btn
btn-primary"><a class="btn" data-dismiss="modal">Close</a></div></div>

  </div></div>

2013/11/21 Kamil Neczaj kneczaj@gmail.com

> Yes, you're right, and there is no white background under the form too.
> It's just draw directly on this black one - on full screen.
> 
> 2013/11/21 Adrian-Tudor Panescu notifications@github.com
> 
> > @kneczaj https://github.com/kneczaj In the current version, when you
> > click on add comment you get this:
> > 
> > [image: comments]https://f.cloud.github.com/assets/480478/1591956/16d69304-52b2-11e3-8ef6-60ddda1067bf.png
> > 
> > Clearly, when you close the dialog, the shaded background goes away.
> > 
> > So I guess it's a Boostrap3 issue that the shaded background does not go
> > away.
> > 
> > —
> > Reply to this email directly or view it on GitHubhttps://github.com/jirikuncar/invenio/pull/59#issuecomment-28984763
> > .

jalavik commented:
@kneczaj To fix the exception you posted, you can probably issue a recreation of the database to deal with the schema changes in "workflows" (beware that you will lose everything in the database):

``` bash
$ inveniomanage database recreate
$ inveniomanage demosite create
$ inveniomanage demosite populate
```

I dunno how to explicitly re-create one table from a model though. `Model.metadata.create_all(db)` does not seem to work.

adrianp commented:
@jalavik To create one specific mode do the following in `inveniomanage shell`:

```
from foo.models import bar
from invenio.ext.sqlalchemy import db

bar.__table__.create(db.engine)
```

jalavik commented on invenio/modules/workflows/templates/workflows/hp_index.html:
```diff
@@ -20,4 +20,46 @@
 
-{% endblock %}
\ No newline at end of file
+    <style type="text/css" title="currentStyle">
```
Everything you add here can be removed as it have been moved to hp_maintable.html

jalavik commented on invenio/modules/workflows/templates/workflows/hp_bibmatch_widget.html:
```diff
@@ -12,19 +12,27 @@
     <h1>Bibmatch Widget</h1>
 
     <div class="container">
+<<<<<<< HEAD
```
Here is some leftover merge conflict.

jalavik commented on invenio/modules/workflows/templates/workflows/hp_bibmatch_widget.html:
```diff
@@ -12,19 +12,27 @@
+        <div class="row">
+            <div class="col-md-12">
+                <div class="row">
+                    {% set widget = widget() %}
```
You can remove this line (set widget part)

jalavik commented on invenio/modules/workflows/static/js/workflows/DT_bootstrap.js:
```diff
@@ -0,0 +1,159 @@
+/* Set the defaults for DataTables initialisation */
```
We will instead change Makefile.am to use DT_bootstrap from version 3.

See the install-jquery-plugins part in Makefile.am. Right now its hardcoded 2, you can change that to 3. Then you can remove this file.
